### PR TITLE
Create project and make it possible to listen to the song

### PR DESCRIPTION
**Pull request by artsiom-streetshares**
**Task:** Implement backend and frontend to stream predefined song.

**Solution steps:**

1. Create empty Django project.
2. Create empty React application.
3. Set up Docker Compose for the project.
4. Add pre-commit hooks (black, pylint, mypy).
5. Create Github action for Docker Compose build.
6. Add RestAPI for song on backend.
7. Add component to stream predefined song on frontend.
8. Set up Swagger for generate API.

### CODE REVIEW

**Review by artsiom-streetshares — COMMENTED** [review]
pyc files got commited

File: backend/simpleMusicService/models.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django.db import models
+
+
+class Song(models.Model):
+    title = models.CharField(max_length=50)
+    artist = models.CharField(max_length=50)
```
Comment (artsiom-streetshares) [inline backend/simpleMusicService/models.py:6]: that's fine for inital proof of concept but later on you'll want to have artists to titles as many to many

File: backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM python:3.8
+
+ENV PYTHONPATH=${PYTHONPATH}:${PWD}
+
+COPY . .
+
+RUN pip3 install poetry
```
Comment (artsiom-streetshares) [inline backend/Dockerfile:7]: recommended way of installing poetry is with script https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions 
installing with pip has a drawback of installing poetry dependencies in your env (we want only our application dependencies)

File: backend/simpleMusicService/admin.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from django.contrib import admin
```
Comment (artsiom-streetshares) [inline backend/simpleMusicService/admin.py:1]: python package names should be snake case like `simple_music_service`

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+services:
+  db:
+    image: postgres
+    environment:
+      - POSTGRES_DB=test
+      - POSTGRES_USER=admin
+      - POSTGRES_PASSWORD=admin
+    ports:
+      - "8080:5432"
```
Comment (artsiom-streetshares) [inline docker-compose.yaml:11]: 8080 is usually for web servers

Comment (artsiom-streetshares) [reply docker-compose.yaml:11]: not a big deal but I would use 5432 on host as well, to make it less surprising

File: frontend/src/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById('root')
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (artsiom-streetshares) [inline frontend/src/index.js:17]: it suggests to pass log function to it

File: frontend/src/style/SongPlayer.css [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+.song-player .artist {
+    text-align: center;
+    color: white;
+}
+
+.song-player .player-component{
+    position: absolute;
+    bottom: 0;
+    width: 100%;
+}
```
Comment (artsiom-streetshares) [inline frontend/src/style/SongPlayer.css:26]: endoffile

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    volumes:
+      - ./data/db:/var/lib/postgresql/data
+
+  backend:
+    build: backend/
+    command: bash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
+    volumes:
+      - .:/code
```
Comment (artsiom-streetshares) [inline docker-compose.yaml:19]: you map current dir to code directory, but runs server on default one. so not sure it does what you expect

**Comment by artsiom-streetshares** [comment]
great job! let's demo, then try deploying to aws (frontend to S3 and backend to elastic beanstalk). 

also at some point I would suggest putting frontend code in it's own repo

File: docker-compose.yaml (hunk 4 above)
Comment (PashkovichKaterina) [reply docker-compose.yaml:11]: I use this port because port 5432 is used on my computer

**Pull request merged**